typings/index.ts

```diff
@@ -25,4 +25,22 @@ client.on('messageReactionRemoveAll', async message => {
   console.log(`messageReactionRemoveAll - content: ${message.content}`);
 });
 
+// These are to check that stuff is the right type
+declare const assertIsMessage: (m: Promise<Message>) => void;
+declare const assertIsMessageArray: (m: Promise<Message[]>) => void;
+
+client.on('message', ({ channel }) => {
+  assertIsMessage(channel.send('string'));
+  assertIsMessage(channel.send({}));
+  assertIsMessage(channel.send({ embed: {} }));
+  assertIsMessage(channel.send({ another: 'property' }, {}));
```

Comment:
Only remark I have is that two valid overloads (`Attachment` and `MessageEmbed` as first parameter) are not tested here. Maybe it's covered by the others, but I want to make sure 👍 

LGTM otherwise.

Reply:
I added some more tests for `MessageAdditions`.